### PR TITLE
Fix Server Message Packet

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Thanks Ino and Eden!

### CODE REVIEW

File: src/map/packets/server_message.cpp [hunk 1]
```diff
@@ -34,24 +34,23 @@ CServerMessagePacket::CServerMessagePacket(const string_t& message, int8 languag
...
+        ref<uint32>(0x10) = message_offset;    // Message Offset
+        ref<uint32>(0x14) = (uint32)sndLength; // Message Length
 
-        memcpy((data + (0x18)), message.c_str() + message_offset, sndLength);
+        memcpy((data + (0x18)) , message.c_str() + message_offset, sndLength);
 
         auto textSize = (uint8)(sndLength + sndLength % 2);
-        // TODO: Verify the below math
-        this->setSize(((((0x14 + textSize) + 4) >> 1) & 0xFE) * 2);
+        this->size = ((((0x14 + textSize) + 4) >> 1) & 0xFE);
```
Comment (claywar) [inline src/map/packets/server_message.cpp]: Remember that this->size (previous implementation) was 16bit values, not by byte.  This line should most likely not change.

Comment (zach2good) [reply src/map/packets/server_message.cpp]: Quite right

**Comment by zach2good** [comment]
Tested now, the annoying 'g' at the end of the server message is now gone 🙌

**Review by claywar — APPROVED** [review]

**Comment by zach2good** [comment]
Tried a bunch of RUN abilities, still works

**Pull request merged**